hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/ControllerManagementTest.java

```diff
@@ -592,16 +587,13 @@ public void findOrRegisterTargetIfItDoesNotExistDoesntRetryWhenEntityAlreadyExis
         when(mockTargetRepository.findOne(any())).thenReturn(Optional.empty());
         when(mockTargetRepository.save(any())).thenThrow(EntityAlreadyExistsException.class);
 
-        try {
-            controllerManagement.findOrRegisterTargetIfItDoesNotExist("1234", LOCALHOST);
-            fail("Expected an EntityAlreadyExistsException to be thrown!");
-        } catch (final EntityAlreadyExistsException e) {
-            verify(mockTargetRepository, times(1)).findOne(any());
-            verify(mockTargetRepository, times(1)).save(any());
-        } finally {
-            // revert
-            ((JpaControllerManagement) controllerManagement).setTargetRepository(targetRepository);
-        }
+        assertThatExceptionOfType(EntityAlreadyExistsException.class)
+                .as("Expected an EntityAlreadyExistsException to be thrown!")
+                .isThrownBy(() -> controllerManagement.findOrRegisterTargetIfItDoesNotExist("1234", LOCALHOST));
+        verify(mockTargetRepository, times(1)).findOne(any());
+        verify(mockTargetRepository, times(1)).save(any());
+        // revert
+        ((JpaControllerManagement) controllerManagement).setTargetRepository(targetRepository);
```

Comment:
I think the `finally` block is still required. If one of the `verify(..)` statements is failing `targetRepository` won't be reverted. Thus further tests are affected  / might be failing due to a wrong mock. This is hard to find

Reply:
Done. The removed finally blocks are re-added again.